src/gui/qgstemporalcontrollerwidget.cpp

```diff
@@ -281,8 +281,13 @@ void QgsTemporalControllerWidget::togglePause()
 
 void QgsTemporalControllerWidget::updateTemporalExtent()
 {
-  QgsDateTimeRange temporalExtent = QgsDateTimeRange( mStartDateTime->dateTime(),
-                                    mEndDateTime->dateTime() );
+  // TODO - consider whether the overall time range set for animations should include the end date time or not.
+  // (currently it DOES include the end date time).
```

Comment:
Not sure if this comment is appropriate/correct?

Actually to me it is so clear what exactly is shown in the slider:
- is it the maximum extent of the DATA (so including the duration of the last feature(s))
- is it the steps that you can go over with the controller?

Does the limit-open/closed discussion have anything to do with this? I don't think so??

Also note that having layers with features with different interval durations there will always be the case where you cannot see features when you are on the end of the slider.

Reply:
> Does the limit-open/closed discussion have anything to do with this? I don't think so??

No, that's why it's just a to-do right now. I don't think the change would have any real world impact, but I'm not confident to go ahead and make it here.